modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/webpack_federation.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+Liferay = window.Liferay || {};
+
+Liferay.Webpack = {
+	Container: {},
+	SharedScope: {},
+};
+
+Liferay.require = (moduleName) =>
```

Comment:
This also feels way to exposed, doesn't it? Since it's only meant to be used by tooling, I'd hide it as much as possible to avoid our past miscues with `Liferay.Loader.require` which is eventually used and abused as much as possible.

Reply:
I first put this inside `Webpack` but [as we said it is more generic and webpack is just one implementation](https://github.com/liferay-frontend/liferay-portal/pull/527#discussion_r528657965) decided to promote it outside of `Webpack`.

Maybe we can leave it here but name it something like `__require` or `__require_dont_use_this_or_you_will_be_fired` like they do in React :sweat_smile: 

Would that be enough? In the end you know that, no matter what we do to hide it, someone will use it :sweat_smile: . So, to me at least, cueing that it should not be used should be enough. But if we can come up with another "more secret" place, it would be great :thinking: 

We can even inline this in the taglibs (it is not so much code) in the final solution. If you think how webpack does it, it puts a code similar to this one in every runtime in every bundle, so we can go that way too.

Reply:
> We can even inline this in the taglibs (it is not so much code) in the final solution. If you think how webpack does it, it puts a code similar to this one in every runtime in every bundle, so we can go that way too.
 
I don't dislike this idea 😂

Reply:
I would say the same we've said above: maybe leave this like it is now, and decide later, when most of the code has been moved to webpack 5...

Reply:
In fact, we can use the same technique for other internal code, maybe... I mean, moving it to `ScriptData` or some other centralized place in the server and write it just once as a <script> tag or something like that...